### DISCUSSION TITLE
Question about connecting

### DISCUSSION THREAD

**Discussion by Skaveny** [hassio-addons/addon-vscode#375] [Q&A] [answered]
Hello I’m wondering can I use this app to connect to it? https://servediter.app/

is the server setup so it can revive connections from code editors? I’m using iPad

**Comment by frenck** [comment, marked as answer]
That won't be possible, as this add-on runs in the frontend of Home Assistant via Ingress.

**Reply by Skaveny** [reply]
Oki Thank you for the quick answer and thank you for the fantastic work you do :)

**Reply by sticilface** [reply]
Does this also mean that it is not possible to connect to VScode server outside the HA GUI?  Looking at the logs 
```
[2022-09-07T14:55:17.746Z] info  Using config file ~/.config/code-server/config.yaml
[2022-09-07T14:55:17.746Z] info  HTTP server listening on http://0.0.0.0:1337/ 
```
Suggests that `http://HAip:1337/` should connect to the VScode instance but it doesn't work.  I suspect as it is set to only accept localhost.

**Reply by sticilface** [reply]
my reason for this is that multiple open windows of home assistant appear to sync, so it is really annoying if I want to keep a page of HA open on node-red, vscode server and a room for example.

**Comment by frenck** [comment]
> Does this also mean that it is not possible to connect to VScode server outside the HA GUI?

No it is not possible.

> Suggests that http://HAip:1337/ should connect to the VScode instance but it doesn't work. I suspect as it is set to only accept localhost.

It is not, as it is containerized / an isolated network environment.

> my reason for this is that multiple open windows of home assistant appear to sync, so it is really annoying if I want to keep a page of HA open on node-red, vscode server and a room for example.

That is not correct and possible. Seems like you are experiencing some other issue in that case.

**Reply by sticilface** [reply]
ok thanks.  
I've just realised that the multiple open windows syncing is a "feature" if you can call it that of safari when you pin a webpage.
